src/time_series_metadata_store.jl

```diff
@@ -78,6 +79,37 @@ function _needs_migration(db::SQLite.DB)
     return "time_series_uuid" in _list_columns(db, METADATA_TABLE_NAME)
 end
 
+function _apply_horizon_type_fix_if_needed(db::SQLite.DB)
+    # Version <= 2.4.1 had a bug where horizon_ms was mistakenly written to the db
+    # as a Serialization.serialize(Dates.Millisecond) instead of an integer.
+    # This function can be removed in in 4.0 when we no longer support deserializing
+    # systems from 2.x.
+    partial_query = """
+        SELECT horizon_ms
+        FROM time_series_associations
+        WHERE horizon_ms IS NOT NULL
+        LIMIT 1
+    """
+    vals = Tables.columntable(SQLite.DBInterface.execute(db, partial_query)).horizon_ms
+    if !isempty(vals) && vals[1] isa Dates.Period
+        query = """
+            SELECT id, horizon_ms
+            FROM $ASSOCIATIONS_TABLE_NAME
+            WHERE horizon_ms IS NOT NULL
+        """
+        update_query = "UPDATE $ASSOCIATIONS_TABLE_NAME SET horizon_ms = ? WHERE id = ?"
+        stmt = SQLite.Stmt(db, update_query)
+        SQLite.transaction(db) do
+            for row in Tables.rowtable(SQLite.DBInterface.execute(db, query))
+                if row.horizon_ms isa Dates.Period
+                    params = (row.horizon_ms.value, row.id)
+                    SQLite.DBInterface.execute(stmt, params)
```

Comment:
Should we be using executemany here?

Reply:
DBInterface.jl says this for executemany:
```
By default, each set of parameters will be looped over and DBInterface.execute will be called for each. 
```
https://juliadatabases.org/DBInterface.jl/dev/#DBInterface.executemany

So, I don't see any value in building an intermediate vector. This seems most natural. I'm open to a counter opinion.